onnx/defs/data_type_utils.h

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#ifndef ONNX_DATA_TYPE_UTILS_H
+#define ONNX_DATA_TYPE_UTILS_H
+
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "onnx/onnx.pb.h"
+
+namespace onnx
+{
+    // String pointer as unique TypeProto identifier.
+    typedef const std::string* DTYPE;
+
+    namespace Utils
+    {
+        // Simple class which contains pointers to external string buffer and a size.
+        // This can be used to track a "valid" range/slice of the string.
+        // Caller should ensure StringRange is not used after external storage has
+        // been freed.
+        class StringRange
```

Comment:
I know it's a bit picky, but do we really need it for a fairly simple parser? Maybe we should just use the plain c++ apis for now

Reply:
@dzhulgakov , I feel it's needed. the current tensor and sparse tensor it fairly simple. We will have more types probably, am I right? at least to support onnx-ml. I don't really want to have two piece of codes to support onnx and onnx-ml. Make sense please?